src/Marten/Internal/Sessions/AmbientTransactionLifetime.cs

```diff
@@ -49,90 +68,253 @@ public void Dispose()
     {
         if (OwnsConnection)
         {
-            Connection?.Close();
-            Connection?.Dispose();
+            _connection?.Close();
+            _connection?.Dispose();
         }
     }
 
     public void Apply(NpgsqlCommand command)
     {
         BeginTransaction();
-        command.Connection = Connection;
+        command.CommandTimeout = CommandTimeout;
+        command.Connection = _connection;
```

Comment:
**QUESTION:** Shouldn't we use `Connection` here or ensure that it's not null?

Reply:
The call to `BeginTransaction()` already makes sure the connection is open